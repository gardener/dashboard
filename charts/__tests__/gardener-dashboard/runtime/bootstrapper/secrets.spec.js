//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const yaml = require('js-yaml')
const { helm, helper } = fixtures
const { getPrivateKey, getCertificate } = helper

const renderTemplates = helm.renderBootstrapperRuntimeTemplates

describe('gardener-dashboard-terminal-bootstrapper', () => {
  let values
  let templates

  beforeEach(() => {
    values = {
      global: {
        bootstrapper: {
          enabled: true
        }
      }
    }
    templates = []
  })

  describe('secret-kubeconfig', function () {
    beforeEach(() => {
      templates.push('secret-kubeconfig')
    })

    it('should not render the template', async function () {
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [kubeconfigSecret] = documents
      expect(kubeconfigSecret).toBe(null)
    })

    it('should render the template', async function () {
      values.global.bootstrapper.kubeconfig = 'apiVersion: v1'
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [kubeconfigSecret] = documents
      expect(kubeconfigSecret).toMatchSnapshot({
        data: {
          'kubeconfig.yaml': expect.any(String)
        }
      })
      const kubeconfig = yaml.load(helper.decodeBase64(kubeconfigSecret.data['kubeconfig.yaml']))
      expect(kubeconfig).toMatchSnapshot('kubeconfig.yaml')
    })
  })

  describe('secret-apiserver-tls', () => {
    const tlsKey = getPrivateKey('tls.key')
    const tlsCrt = getCertificate('tls.crt')

    beforeEach(() => {
      templates.push('secret-apiserver-tls')
    })

    it('should render the templates', async function () {
      values.global.terminal = {
        bootstrap: {
          disabled: false,
          gardenTerminalHostDisabled: false,
          gardenTerminalHost: {
            apiServerTls: {
              key: tlsKey,
              crt: tlsCrt
            }
          }
        }
      }
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [terminalSecret] = documents
      expect(terminalSecret).toMatchSnapshot()
    })
  })
})
