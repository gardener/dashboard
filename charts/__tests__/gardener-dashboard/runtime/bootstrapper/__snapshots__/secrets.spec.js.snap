// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard-terminal-bootstrapper secret-apiserver-tls should render the templates 1`] = `
Object {
  "apiVersion": "v1",
  "data": Object {
    "tls.crt": "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCmRHeHpMbU55ZEE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0t",
    "tls.key": "LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpkR3h6TG10bGVRPT0KLS0tLS1FTkQgUlNBIFBSSVZBVEUgS0VZLS0tLS0=",
  },
  "kind": "Secret",
  "metadata": Object {
    "labels": Object {
      "app.kubernetes.io/component": "terminal-bootstrapper",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-runtime-0.1.0",
    },
    "name": "garden-host-cluster-apiserver-tls",
    "namespace": null,
  },
  "type": "kubernetes.io/tls",
}
`;

exports[`gardener-dashboard-terminal-bootstrapper secret-kubeconfig should render the template 1`] = `
Object {
  "apiVersion": "v1",
  "data": Object {
    "kubeconfig.yaml": Any<String>,
  },
  "kind": "Secret",
  "metadata": Object {
    "labels": Object {
      "app.kubernetes.io/component": "terminal-bootstrapper",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-runtime-0.1.0",
    },
    "name": "gardener-dashboard-terminal-bootstrapper-secret-kubeconfig",
    "namespace": "garden",
  },
  "type": "Opaque",
}
`;

exports[`gardener-dashboard-terminal-bootstrapper secret-kubeconfig should render the template: kubeconfig.yaml 1`] = `
Object {
  "apiVersion": "v1",
}
`;
